### PR TITLE
n5010 support

### PR DESCRIPTION
**Pull request by hundeboll-silicom**
I've concluded that keeping the n5010 support in feature branch is just added maintenance for little gain, so let's instead merge the support to master.

### CODE REVIEW

File: tools/libboard/board_n5010/board_n5010.c [hunk 1]
```diff
@@ -0,0 +1,266 @@
+// Copyright(c) 2019-2020, Intel Corporation
```
Comment (tswhison) [inline tools/libboard/board_n5010/board_n5010.c:1]: Could we add some unit tests for this new board module under tests/board/?

Comment (hundeboll-silicom) [reply tools/libboard/board_n5010/board_n5010.c:1]: Done

File: tests/board/test_board_n5010.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
+// Copyright(c) 2019-2020, Intel Corporation
+// Copyright(c) 2021, Silicom Denmark A/S
```
Comment (tswhison) [inline tests/board/test_board_n5010.cpp:2]: For the dual copyright files, I posed the question through the ranks and to legal, and their feedback indicated that the copyright statements should state something like the following.  Is this ok?
```bash
## Original work Copyright(c) 2019-2020, Intel Corporation
## Modifications Copyright(c) 2020-2021, Silicom Denmark A/S
```

File: tools/libboard/board_n5010/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,40 @@
+## Copyright(c) 2019-2020, Intel Corporation
```
Comment (tswhison) [inline tools/libboard/board_n5010/CMakeLists.txt:1]: Please see note pertaining to dual-copyright files.

File: tools/libboard/board_n5010/board_n5010.c (hunk 1 above)
Comment (tswhison) [inline tools/libboard/board_n5010/board_n5010.c:1]: Please see note about dual-copyright files.

File: tools/libboard/board_n5010/board_n5010.h [hunk 4]
```diff
@@ -0,0 +1,125 @@
+// Copyright(c) 2019-2020, Intel Corporation
```
Comment (tswhison) [inline tools/libboard/board_n5010/board_n5010.h:1]: Please see note about dual-copyright files.

File: tests/board/test_board_n5010.cpp (hunk 2 above)
Comment (hundeboll-silicom) [reply tests/board/test_board_n5010.cpp]: Fixed

File: tools/libboard/board_n5010/CMakeLists.txt (hunk 3 above)
Comment (hundeboll-silicom) [reply tools/libboard/board_n5010/CMakeLists.txt]: Fixed

File: tools/libboard/board_n5010/board_n5010.h (hunk 4 above)
Comment (hundeboll-silicom) [reply tools/libboard/board_n5010/board_n5010.h]: Fixed

**Review by tswhison — APPROVED** [review]

**Comment by hundeboll-silicom** [comment]
@michelleyho @r-rojo Please review!

**Review by anandaravuri — APPROVED** [review]

**Review by michelleyho — APPROVED** [review]

**Pull request merged**